### PR TITLE
nest: fix Python & Darwin

### PR DESCRIPTION
**Pull request by CyborgPotato**
Removed auto-patch-elf and replaced with find command that moves libraries where they are expected by nest. Auto-patchelf isn't supported on aarch64, so I copied my local derivation fix I've been using.

PyNEST also requires numpy, so that has been added to propagatedbuildinputs.

Also added support for OpenMP on Darwin.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [X] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jiegec** [comment]
Result of `nixpkgs-review pr 217897` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>nest</li>
    <li>nest-mpi</li>
    <li>python310Packages.nest</li>
    <li>python311Packages.nest</li>
  </ul>
</details>

**Review by jiegec — APPROVED** [review]

File: pkgs/applications/science/biology/nest/default.nix [hunk 1]
```diff
@@ -37,24 +40,36 @@ stdenv.mkDerivation rec {
...
     pkg-config
-    autoPatchelfHook
   ];
 
   buildInputs = [
     gsl
     readline
     libtool # libltdl
+    boost
+    findutils
```
Comment (wegank) [inline pkgs/applications/science/biology/nest/default.nix:50]: Should `findutils` go to `nativeBuildInputs`?

Comment (DavidCromp) [reply pkgs/applications/science/biology/nest/default.nix]: Yes it should thank you for pointing it out, it has now been corrected.

**Review by jiegec — APPROVED** [review]

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 217897` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>nest</li>
    <li>nest-mpi</li>
    <li>python310Packages.nest</li>
    <li>python311Packages.nest</li>
  </ul>
</details>

**Pull request merged**